### PR TITLE
Document soundex function

### PR DESCRIPTION
**Pull request by Damans227**
**What type of PR is this:**
Updates the documentation

**What this PR does / why we need it:**
To document soundex function

**Which issue(s) this PR fixes:**
Document soundex function [#7668](https://github.com/trinodb/trino/issues/7668)

**Special notes for your reviewer:**
N/A

### CODE REVIEW

File: docs/src/main/sphinx/functions/string.rst [hunk 1]
```diff
@@ -134,6 +134,21 @@ String functions
 
     Removes trailing whitespace from ``string``.
 
+.. function:: soundex(char) -> string
+
+   ``soundex`` returns a character string containing the phonetic representation of ``char``.
+    This function is typically used to evaluate the similarity of two expressions phoenetically,
```
Comment (martint) [inline docs/src/main/sphinx/functions/string.rst:140]: Typo: phoenetically

File: docs/src/main/sphinx/functions/string.rst [hunk 2]
```diff
@@ -134,6 +134,21 @@ String functions
 
     Removes trailing whitespace from ``string``.
 
+.. function:: soundex(char) -> string
+
+   ``soundex`` returns a character string containing the phonetic representation of ``char``.
+    This function is typically used to evaluate the similarity of two expressions phoenetically, 
+    i.e how the string sounds when spoken.
```
Comment (martint) [inline docs/src/main/sphinx/functions/string.rst:141]: `i.e.,`

**Review by martint — COMMENTED** [review]
Thanks! Just a couple of minor comments.

**Review by Damans227 — COMMENTED** [review]
Fixed the grammar issues. Sorry, will do better next time.

File: docs/src/main/sphinx/functions/string.rst [hunk 3]
```diff
@@ -134,6 +134,21 @@ String functions
 
     Removes trailing whitespace from ``string``.
 
+.. function:: soundex(char) -> string
+
+   ``soundex`` returns a character string containing the phonetic representation of ``char``.
+    This function is typically used to evaluate the similarity of two expressions phonetically,
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/string.rst:140]: It is typically used to evaluate the similarity of two expressions phonetically,  that is how the string sounds when spoken::

File: docs/src/main/sphinx/functions/string.rst [hunk 4]
```diff
@@ -134,6 +134,21 @@ String functions
 
     Removes trailing whitespace from ``string``.
 
+.. function:: soundex(char) -> string
+
+   ``soundex`` returns a character string containing the phonetic representation of ``char``.
+    This function is typically used to evaluate the similarity of two expressions phonetically, 
+    i.e., how the string sounds when spoken.
+
+    Here is an example to illustrate the ``soundex`` function::
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/string.rst:143]: Drop this redundant sentence

File: docs/src/main/sphinx/functions/string.rst [hunk 5]
```diff
@@ -134,6 +134,21 @@ String functions
...
+
+        SELECT * FROM nation WHERE SOUNDEX(name)  = SOUNDEX('CHYNA');
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/string.rst:145]: Maybe break into multiple lines and only return name?

```
SELECT name 
FROM nation
WHERE SOUNDEX(name)  = SOUNDEX('CHYNA');
```

**Review by mosabua — CHANGES_REQUESTED** [review]
Thanks for helping out. Just a few tweaks. 

Note that we avoid i.e. as part of the Google style https://developers.google.com/style/word-list#letter-i

File: docs/src/main/sphinx/functions/string.rst (hunk 4 above)
Comment (Damans227) [reply docs/src/main/sphinx/functions/string.rst]: Done!

File: docs/src/main/sphinx/functions/string.rst (hunk 3 above)
Comment (Damans227) [reply docs/src/main/sphinx/functions/string.rst]: Fixed it, apologies for this. Will do better next time.

File: docs/src/main/sphinx/functions/string.rst (hunk 5 above)
Comment (Damans227) [reply docs/src/main/sphinx/functions/string.rst]: Done!

**Comment by Damans227** [comment]
@mosabua @martint Just putting this on your radar to review and merge the new changes I've made. Cheers!

**Review by mosabua — APPROVED** [review]

**Comment by mosabua** [comment]
Looks great now. Please squash the commits into one commit.

**Comment by mosabua** [comment]
@Damans227 let us know if you have problems with squashing the commits.

**Comment by Damans227** [comment]
> @Damans227 let us know if you have problems with squashing the commits.

Thanks @mosabua I have tried squashing the commits just now and it seem to have worked successfully. Please review the changes and share your feedback.

**Pull request merged**